joplin/pages/base_page/migrations/0002_add_pageviewpermissions.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+def add_pageviewpermissions(apps, schema_editor):
+    # Get model
+    JanisBasePage = apps.get_model('base_page.JanisBasePage') #todo: check this
+    PageViewRestriction = apps.get_model('wagtailcore.PageViewRestriction')
+
+    all_pages = JanisBasePage.objects.all()
+
+    # https://blog.etianen.com/blog/2013/06/08/django-querysets/
+    for page in all_pages.iterator():
+        pvr = PageViewRestriction.objects.create(
+            page=page,
+            restriction_type='groups',
```

Comment:
oddly enough, when I put `PageViewRestriction.GROUPS` just as it is on line 99 of new_page_from_modal above, the migrations errored. does anyone know why?

Reply:
Models are weird when used inside migration files. It's not the same object as a real PageViewRestriction. In fact, it's a __fake__ PageViewRestriction:

![Screen Shot 2020-06-04 at 10 14 03 AM](https://user-images.githubusercontent.com/14187277/83775268-56128100-a64c-11ea-95d8-998dfb7045ed.png)

So not all attributes and methods are available. Why some things are restricted inside migration files, I don't know. I just trust that django knows what it's doing, and I copy techniques that other projects have used in their migration files to work around limitations.